crates/core/src/host/wasmer/wasmer_module.rs

```diff
@@ -377,10 +389,23 @@ impl WasmerInstance {
                 Err(crate::util::string_from_utf8_lossy_owned(errmsg.into()).into())
             })
         });
+
+        // Close the `wasm_execution_duration` span and collect the WASM execution time.
+        let wasm_duration = self.env.as_mut(store).timing_end_reducer();
+
+        // Close the `execution_duration` span and collect the overall execution time.
+        // Unlike `wasm_duration`, this includes time spent in host functions,
+        // e.g. evaluating queries.
+        let duration = start.elapsed();
+
+        // Clear buffers and timing spans to avoid leaking data across reducer runs.
         self.env.as_mut(store).buffers.clear();
+        self.env.as_mut(store).iters.clear();
+        self.env.as_mut(store).timing_spans.clear();
```

Comment:
I'm now wondering why you are doing this here but not other places that were calling `self.env.as_mut(store).buffers.clear()`, specifically in `call_describer`. I guess that we just don't care about the timing in that path right now?

Also, why `self.env.as_mut(store).iters.clear()` is now being called here now.

IMO the lack of encapsulation between wasm_instance_env and instance_env makes this challenging to understand.

Reply:
Ah, I didn't mean to commit this change. At some point we should investigate what state we do and don't persist between calls, as I think we're probably leaking some of it that we don't want to, but I think that's out of scope.